protostar/testing/environments/execution_environment.py

```diff
@@ -38,6 +38,9 @@ async def perform_execute(
         **kwargs: Any,
     ) -> PerformExecuteResult:
         try:
+            assert (
```

Comment:
If it does not work for every ExecutionEnvironment i would move it to a proper child-class

Reply:
I restructured the classes